src/main/java/com/univ/lille/copainsderoute/platine/service/UserService.java

```diff
@@ -116,13 +129,72 @@ private boolean userWithProfilePic(int userId) {
     }
 
     public String getUserProfilePicLocation(User user) {
-        if(userWithProfilePic(user.getId())) {
+        if (userWithProfilePic(user.getId())) {
             return USER_PATH.concat(String.valueOf(user.getId())).concat(PROFILE_PIC_PATH);
         }
         return null;
     }
 
     public List<FriendsRequestResponseDTO> createFriendList(List<Friends> friends) {
-        return friends.stream().map(f -> new FriendsRequestResponseDTO(f, getUserProfilePicLocation(f.getSender()), getUserProfilePicLocation(f.getAdded()))).toList();
+        return friends.stream().map(f -> new FriendsRequestResponseDTO(f, getUserProfilePicLocation(f.getSender()),
+                getUserProfilePicLocation(f.getAdded()))).toList();
     }
+
+    public void resetPassword(String email, String newPassword) throws UserNotFoundException {
+        User user = userRepository.findByEmail(email);
+        if (user == null) {
+            throw new UserNotFoundException();
+        }
```

Comment:
Avec Optional tu pourras directement faire 
```suggestion
        User user = userRepository.findByEmail(email).orElseThrow(UserNotFoundException::new);
```